### ISSUE TITLE
add the color blue to find the bobber in lava pools

### ISSUE THREAD

**Issue by zippoking** [julianperrott/FishingFun#17]
Hi there,

the bot is just amazing and its so easy to use. So big thank you! 
Unfortunately it´s not working in lava pools, because of too much red color on the screen.

However there are other colors which stick out and that is blue and white-ish, because of the blue feather on the bobber or the white-ish bobber itself.  If you look at this picture i think its pretty clear what i´m thinking of.

![image](https://user-images.githubusercontent.com/18005338/98360680-878fde00-202a-11eb-951d-6838867a1509.png)

So my question to is, if there is any way for you to implement a button, which i/we can activate to tell the bot to look for blue/white-ish color instead of red?

I specificially need this for the https://www.wowhead.com/achievement=9455/fire-ammonite-angler archievment.

Keep up the good work!
Thanks in advance

Zippoking

**Comment by Grootloops33** [comment]
Bump!

I'm having the exact same issue, trying to farm for Volatile fire.

**Comment by julianperrott** [comment]
Hi, take a look at this file https://github.com/julianperrott/FishingFun/blob/master/Source/FishingFunBot/Bot/PixelClassifier.cs

Let me know if changing the function to this works? Basically I have swapped red for blue.

```
  public bool IsMatch(byte red, byte green, byte blue)
        {
            return isBigger(blue, green) && isBigger(blue, red) && areClose(red, green);
        }
```

**Comment by julianperrott** [comment]
I have added a new configuration option for watching the blue feather. let me know how you get on.

**Comment by zippoking** [comment]
You are a god!!!!!

First of all thank you for the reply and for the quick fix/new feature.

I just tested the bot and it works very good in lava pools now.
Only thing i noticed is, that sometimes the bot doesn´t loot even though it recognized the bobber movement.
Maybe i need to play around with color settings or character positioning. I will test more on the weekend.

I have another question more off topic:
Since blizzard "banned" multiboxers recently, they probably have now a detection method to find the software tools which send the keystrokes to multiple wow-clients. Do you think this has any affect on your bot? Is your bot "more" detectable now?


Once again, thank you so much!
Best regards
Zippoking

**Comment by julianperrott** [comment]
Does it take too long to click the bobber or just not click at all? There is a setting now to allow the delay before clicking to be longer or shorter, if it is too long it will miss the fish, is that what is happening ?

I don't know what blizz are going to do, perhaps they will look for specific apps that are running and warn/temp ban people using them. I guess it won't be a perma ban first, but there is always a risk! They will be able to see the source and origin of mouse and keyclicks and that it is not from hardware, https://github.com/julianperrott/FishingFun/issues/4. I might have a play with https://github.com/jasonpang/Interceptor and see if it is worth trying.

**Comment by zippoking** [comment]
I tested the bot for another hour and it is amazing! i got the achievment by the way:-D.
I changed the setting for the loot delay to a much shorter time and not a single time the bot missed a fish. Just perfect!


Okay, thanks for the info. I keep looking for updates, especially if you get the Interceptor to work.



From my side everything is fixed, so should i close the issue or are you going to do it?
